CodeiumVS/LanguageServer/LanguageServerController.cs

```diff
@@ -231,6 +231,42 @@ public async Task RefactorFunctionAsync(string prompt, string filePath, Function
         await Package.ShowToolWindowAsync(typeof(ChatToolWindow), 0, create: true, Package.DisposalToken);
     }
 
+    public async Task ExplainProblemAsync(string problemMessage, SnapshotSpan span)
+    {
+        ITextSnapshotLine problemLineStart = span.Snapshot.GetLineFromPosition(span.Start);
+        ITextSnapshotLine problemLineEnd = span.Snapshot.GetLineFromPosition(span.End);
+
+        int surroundingLineStart_no = Math.Max(problemLineStart.LineNumber - 10, 0);
+        int surroundingLineEnd_no = Math.Min(problemLineStart.LineNumber + 10, span.Snapshot.LineCount - 1);
```

Comment:
The `surrounding_code_snippet` is 20 lines around the problem, this matches how the vscode extension works, but should we increase it? I've had better results increasing it to the surrounding code block and not just by lines.

Reply:
We can increase it, but we probably shouldn't use the whole code block if it's too long--there's a message length limit imposed by our chat models.